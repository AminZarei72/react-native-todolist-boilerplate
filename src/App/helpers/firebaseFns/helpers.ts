/* ====================================================== */
/**
 * specific fetch for firebase server
 * @param args 
 */
export async function fbFetch(args: {
    url: string,
    token?: string,
    firebase_host?: string,
    finalBody?,
    method: 'DELETE' | 'GET' | 'PATCH' | 'POST',
}) {
    try {
        const headers = {}
        if (args.token) headers['Authorization'] = `Bearer ${args.token}`
        const options = {
            method: args.method,
            headers: headers,
        }
        if (args.finalBody) options['body'] = JSON.stringify(args.finalBody)
        const res: Response = await fetch(args.url, {
            ...options,
        })
            .catch(e => {
                console.error(e)
                return null
            })
        if (!res) {
            console.error('couldnt fetch!')
            return null
        } 
        const resJson = await res.json() 
        return resJson
    } catch (error) {
        console.log(error)
        return null
    }
}
/* ====================================================== */
export function getFireType(targetValue: any,fb_fs_url:string) {
    if (targetValue === null) {
        return { type: 'nullValue', value: 'NULL_VALUE' }
    }
    else if (typeof targetValue === 'boolean') {
        return { type: 'booleanValue', value: targetValue }
    }
    else if (typeof targetValue === 'number') {
        return { type: 'integerValue', value: Number(targetValue) }
    }
    else if (typeof targetValue === 'string') {
        return { type: 'stringValue', value: targetValue.toString() }
    }
    /* todo:still so many more types plus times  */
    else if (typeof targetValue === 'object') { /* null also is an object but we got it earler */
        if (targetValue.ref) {
            return { type: 'referenceValue', value: `${fb_fs_url}/${targetValue.ref.col}/${targetValue.ref.id}` }
        } else if (Array.isArray(targetValue)) {
            const values = []
            targetValue.map(i => {
                const { type, value } = getFireType(i,fb_fs_url)
                values.push({ [type]: value })
            })
            return { type: 'arrayValue', value: { values: values } }
        }
        else {
            const fields = {}
            for (const key in targetValue) {
                const { type, value } = getFireType(targetValue[key],fb_fs_url)
                fields[key] = { [type]: value }
            }
            return { type: 'mapValue', value: { fields: fields } }
        }
    }
    else {
        throw 'not supported!!!'
    }
}
/* ====================================================== */
 
