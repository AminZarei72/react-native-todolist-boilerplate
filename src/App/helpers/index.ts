export * from './gif'
export * as firebaseFns from './firebaseFns'

// const {
//     imgLoader,
// }=require('./fileLoaders')
// export {
//     imgLoader,
// }
