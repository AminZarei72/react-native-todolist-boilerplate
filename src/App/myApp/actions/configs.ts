import { Platform } from 'react-native'
import * as types from '../types'

export const development = true
/** paths that modules like menu or tabs must be invisible  */
export const pathsToBeInvisible:types.routes[] = [
    '',
    'Init',
    // '/Login',
]

/* ======================================== */
export const host = 'http://localhost'  
const fbAppName='todolist-1'
export const firebase_fireStore_port = 9101
export const firebase_fireStore_url = host + ':' + firebase_fireStore_port + '/v1/projects/'+fbAppName+'/databases/(default)/documents'
 
/* const firebase_host = 'https://firestore.googleapis.com/v1/projects/PROJECT_HERE/databases/(default)/documents' */
/* ======================================== */ 

// //@ts-ignore
// import aboutMeImage from '../media/aminZarei.jpg'
//@ts-ignore
//  defaultBgImg = require('../media/images/bgs/bg1.jpg')
// const aboutMeImage=require('../media/images/aminZarei.jpg')
// export {
//     assetDir_a,
//     defaultMarketerImg,
//     defaultBgImg,
//     defaultMarketerImgPlain,
//     imageComponentPlatformQuirk,
//     defaultBgPlain,
//     // aboutMeImage, 
// }
/* ========================================== */
/* texts */
export const appName = 'rn-boilerplate'
export const aboutUs = {
    firstName: 'Amin',
    lastname: 'Zarei',
    number: 'https://github.com/aminZarei72',
    email: 'AminZarei.work@gmail.com',
    title: 'aboutUs',
    subtitle: 'a zero-config react-redux boilerplate for web and native and ready to publish.',
    descreiption: `
    a todoList app ready-to-publish boilerplate for rn web and android with redux,cypress,ts-firebase-rules,detox,rntl,paper and more
        - ts-firebase-rules: which let you write firebase *.rules files with typescript 
        - cypress/detox(e2e) and basic example
        - react-native-testing-library(unit-test) and basic example
        - Paper and useful Icons libraries
        - react-router
        - ...
        visit https://github.com/aminZarei72/react-native-todolist-boilerplate for more detail
    `,
    aboutMe: `
    i'm a fullstack developer who loves programming with web technologies like react-native,
    clouds and testing frameworks which made developing apps more fun and stable.
    https://github.com/aminZarei72/
    `,
}
export const loginTitleMsg = 'wellcome to ' + appName
export const initilizingMsg = 'initializing...' 
