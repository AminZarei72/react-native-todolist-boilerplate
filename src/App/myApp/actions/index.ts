
export * as configs from './configs'
export * from './init'
export * from './todos/fns'

