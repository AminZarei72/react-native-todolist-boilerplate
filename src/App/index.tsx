import React from 'react'
import { BackHandler, I18nManager, Dimensions, Text, View, Platform } from 'react-native'
import { Provider } from 'react-redux'
import store from './utils/store'
import { DefaultTheme, DarkTheme, configureFonts, Provider as PaperProvider, Portal, ProgressBar } from 'react-native-paper'
import Router from './utils/Router'
import EsEnvUiContentManager from './myApp/pages/EsEnvUiContentManager'
import Toast from './myApp/pages/Toast'
// import * as configs from './myApp/actions/configs'
// import * as types from './myApp/actions/types' 
import * as myApp from './myApp/actions'
import Popup from './myApp/pages/Popup'
import EsTabs from './myApp/pages/EsTabs'
import PlusMenuBtn from './myApp/pages/PlusMenuBtn'
import TopBar from './myApp/pages/TopBar'
import LeftMenu from './myApp/pages/LeftMenu'
import Spinner from './myApp/pages/Spinner'
import * as helpers from './helpers'
/* --------------------------- */
const theme = {
    ...DefaultTheme,
    colors: {
        ...DefaultTheme.colors,
        // primary: 'tomato',
        // accent: 'yellow',
    },
    // fonts: configureFonts(fontConfig),
}
export default class App extends React.Component {
    /* --------------------------- */
    constructor(props) {
        super(props)
        BackHandler.addEventListener('hardwareBackPress', () => {
            const c_r = helpers.getRouteName()
            if ([ '/'].includes(c_r)) {
                helpers.showPupop({
                    title: 'exit',
                    content: 'exit the app?',
                    type: 'confirm',
                    callbackFn: (e) => {
                        if (e) {
                            try {
                                BackHandler.exitApp()
                            } catch (error) {
                                console.log(error)
                            }
                        }
                    }
                })
            } else {
                helpers.goBack()
            }
            return true
        })
    }
    /* --------------------------- */
    async componentDidMount() {
        helpers.showSpinner({ content: 'please wait for app to initilize...' })
        if (Platform.OS == 'web') {
            Dimensions.addEventListener('change', (asd) => {
                window.location.reload()
            })
        }
        await myApp.initAppAfterComponentDidMount()
        helpers.showSpinner(false)
    }
    /* --------------------------- */
    render() {
        return (
            <Provider store={store} >
                <PaperProvider theme={theme} >
                    <React.Fragment > 
                        <View style={{
                            position: 'absolute',
                            top: 0,
                            left: 0,
                            width: Dimensions.get('window').width,
                            height: Platform.OS == 'web' ? Dimensions.get('window').height : Dimensions.get('window').height - 20,
                            overflow: 'hidden',
                        }}>
                            {/* ----------- */}
                            <View style={{
                                position: 'relative',
                                top: 0,
                                left: 0,
                                width: '100%'
                            }}>
                                <TopBar
                                    on_goBack_cb={() => helpers.goBack()}
                                    on_btn2_clicked={(args) => console.log('on_btn2_clicked')}
                                    on_btn3_clicked={() => console.log('on_btn3_clicked')}
                                    on_activateTopMenu_cb={(newState) => helpers.setStore({ activateTopMenu: newState })}
                                />
                            </View>
                            {/* ----------- */}
                            <View style={{}}>
                                <EsEnvUiContentManager   >
                                    <Router />
                                    {/* <Text>ok</Text> */}
                                </EsEnvUiContentManager>
                            </View>
                            <EsTabs />
                            {/* ----------- */}
                            <Portal>{/* Note: above tags have higher zIndex / portals in the same level have their own zindex */}
                                <PlusMenuBtn
                                    on_btn1_clicked={() => helpers.goto('AddNewTodo')}
                                    on_btn2_clicked={(args) => console.log('on_btn2_clicked')}
                                    on_btn3_clicked={() => console.log('on_btn3_clicked')}
                                    // on_onStateChange_cb={}
                                />
                                <LeftMenu
                                    on_blankSpacePressed_cb={() => helpers.setStore({ activateTopMenu: false })}
                                    on_profilePicturePressed_cb={async (marketerDetailsId) => {
                                        console.log('on_profilePicturePressed_cb')
                                    }}
                                    on_itemsPressed_cb={(page, params?) => {
                                        helpers.goto(page, params ? params : null)
                                        helpers.setStore({ activateTopMenu: false })
                                    }} 
                                />
                            </Portal>
                            <Portal>
                                <Spinner />
                            </Portal>
                            <Portal>
                                <Popup />
                            </Portal>
                            <Portal>
                                <Toast />
                            </Portal>
                            {/* ----------- */}
                        </View>
                    </React.Fragment>
                </PaperProvider>
            </Provider>
        )
    }
    /* --------------------------- */
}

