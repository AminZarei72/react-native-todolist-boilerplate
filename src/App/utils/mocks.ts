

import * as types from '../myApp/types'
/* ======================================= */
export function mock_addNewTodo(): types.todoFormData {
    return {
        title: 'todo1', 
        comments: 'this is a todo that im not going to do it anyway ;) ',
    }
}
/* ======================================= */
