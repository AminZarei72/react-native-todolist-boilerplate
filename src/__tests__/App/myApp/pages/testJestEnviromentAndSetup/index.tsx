import 'react-native'
import * as React from 'react'
// https://callstack.github.io/react-native-testing-library/docs/api/
// import { render, screen, fireEvent, renderWithActualStore, waitForDomChange, cleanup, waitForElement, renderWithActualStore2, getByRole, getByText, } from '../../../../../App/utils/test-utils'
// import { render, screen, fireEvent, renderWithActualStore, waitForDomChange, cleanup, waitForElement, renderWithActualStore2, getByRole, getByText, } from '../../../../../App/utils/test-utils'
import * as mocks from '../../../../../App/utils/mocks'
// import Cmp from '../../../../../App/myApp/pages/Login'
import * as helpers from '../../../../../App/helpers'
import * as configs from '../../../../../App/myApp/actions/configs'
// import {delay} from '../../../../../App/helpers/AzPureFunctions2'
import * as myApp from '../../../../../App/myApp/actions'
import {
  render,
  fireEvent,
  waitForElementToBeRemoved,
  waitFor,
} from '@testing-library/react-native'
import {expect, it, jest, test} from '@jest/globals'
import {Text, TouchableHighlight, View} from 'react-native'
import FakeFlatlist from '../../../../../FakeTestingCmp1'
import FakeLoginForm from '../../../../../FakeTestingCmp2'
// const expect2:any=expect
///////
// expect(getByText(/Click me/i).getAttribute("disabled")).toBe(null)
// expect(getByText(/Click me/i).closest('button')).toHaveAttribute('disabled');
// expect(getByText(/Click me/i).closest('button')).toBeDisabled();
/* ======================================== */
// export function isDisabled(element: Nullable<ReactTestInstance>): boolean {
//   return !!element?.props.onStartShouldSetResponder?.testOnly_pressabilityConfig()?.disabled;
// }
// expect(isDisabled(getByText("Cancel")).toEqual(false);
/* ======================================== */

it('fake a login form', async () => {
  const username = 'hi'
  const password = 'qwerty1234'
  let submittedData = {}
  // @ts-ignore
  const handleSubmit = jest.fn(data => (submittedData = data))
  const {getByText, getByPlaceholderText,debug,toJSON} = render(
    <FakeLoginForm onSubmit={handleSubmit} />,
  )
  const button = getByText(/submit/i)

  await fireEvent.changeText(getByPlaceholderText(/username/i), username)
  await fireEvent.changeText(getByPlaceholderText(/password/i), password)
  fireEvent.press(button)

  expect(submittedData).toEqual({password, username})
  expect(handleSubmit).toHaveBeenCalledWith({password, username})
  expect(handleSubmit).toHaveBeenCalledTimes(1)
  expect(toJSON()).toMatchSnapshot()
  // debug()
})
/* ======================================== */
it('test a flatlist and a scrolling affect ', async () => {
  /* ---------------------------------- */
  const {getByText, getByTestId, debug,toJSON} = render(<FakeFlatlist />)
  getByText(/pizza/i)
  expect(() => getByText(/the impossible burger/i)).toThrow(
    'Unable to find an element with text: /the impossible burger/i',
  ) //intially not shown

  const eventData = {
    nativeEvent: {
      contentOffset: {
        x: 0,
        y: 425,
      },
      contentSize: {
        // Dimensions of the scrollable content
        height: 885,
        width: 328,
      },
      layoutMeasurement: {
        // Dimensions of the device
        height: 469,
        width: 328,
      },
    },
  }

  expect(getByTestId('flat-list')).toBeDefined()
  fireEvent.scroll(getByTestId('flat-list'), eventData)
  // await waitFor(() => expect(getByText(/the impossible burger/i)))
  //  console.log()
  // getByTestId('flat-list')
  await waitForElementToBeRemoved(() => getByText(/loading more dishes/i),{timeout:3000})
  await waitFor(() => expect(getByText(/the impossible burger/i)))
  expect(toJSON()).toMatchSnapshot()
  // debug()

  return
  /* ---------------------------------- */
  // /* ---------------------------------- */
  // const MyComponent = _props => (
  //   <View>
  //     <TouchableHighlight testID={'touchable-asd'}>
  //       <Text>{'Foo'}</Text>
  //     </TouchableHighlight>
  //   </View>
  // )

  // // let aaa=1
  // const pressedCallback = jest.fn(()=>{})
  // const { getByTestId,debug,getByRole,getByText } = render(<MyComponent onPress={pressedCallback} />)
  // debug()
  // // await delay(1000)
  // fireEvent.press(getByText(/Foo/i)) //didnt work
  // // fireEvent.press(getByTestId('touchable-asd')) //didnt work
 
  // // rnfireEvent.press(getByText('Foo').parent)
  // // expect(aaa).toEqual(2) //Expected mock function not to be called but it was called with: ["bar"]
  // expect(pressedCallback).toBeCalled() //Expected mock function not to be called but it was called with: ["bar"]
  // return
  // /* ---------------------------------- */
  /* new functions(just for learning) */
  // expect(innerText.props.style).toMatchObject({color: 'black'})
  // expect(counterText.props.children).toEqual(['Current count: ', 0])
  /* ---------------------------------- */
  /* main */
  // const {queryByText,getAllByTestId,debug,getAllByRole,getByTestId,getByText}=renderWithActualStore2(<Cmp />)
  // expect2(getByText(configs.loginTitleMsg,{exact:true})).toBeDefined()//ok
  // expect(getByTestId(/login-btn/i)).toBeDefined() //does not work!
  /* ---------------------------------- */

  // await delay(1000)
  // debug(queryByText(configs.loginTitleMsg+' asd ').toString())
  // const asd=
  // console.log(asd)
  // debug(getByTestId('login'))
  // console.log(getByTestId('login'))
  // debug(getByTestId('login')).toBeDisabled()
  // debug(getAllByRole('button'))
  // expect(getAllByRole('button')[0]).toBeDisabled()
  // debug()

  // // @ts-ignore
  // expect(getByText(/login/i)).toBeDisabled()
  // expect(getAllByTestId(/login/i)).toBeDisabled()
})
// await waitFor(() => getByText('Banana ready'));
// const detailsScreen = within(getByA11yHint('Details Screen'));
// expect(detailsScreen.getByText('Some Text')).toBeTruthy();
// expect(detailsScreen.getByDisplayValue('Some Value')).toBeTruthy();
// expect(detailsScreen.queryByA11yLabel('Some Label')).toBeTruthy();
// await expect(detailsScreen.findByA11yHint('Some Label')).resolves.toBeTruthy();
/* ======================= */

// const {getByText, getByTestId, getAllByTestId, queryByText} = render(
//   <Home />,
// );
// const input = getByTestId('adder-input');
// const button = getByText('Add');
// fireEvent.changeText(input, 'item0');
// fireEvent.press(button);
// fireEvent.changeText(input, 'item1');
// fireEvent.press(button);
// const item0 = getByText('item0');
// const item1 = getByText('item1');

// expect(item0).toBeDefined();
// expect(item1).toBeDefined();
// fireEvent.press(getAllByTestId('cell-delete')[0]);
// expect(queryByText('item0')).toBeNull();
// const list = getByTestId('list');

/* ======================= */
//mock again and see if this test works again
/* ======================= */

// console.log()
// screen.debug(screen.getByTestId('switch1'))
// fireEvent.click(screen.getByTestId('switch1').querySelector('input'))
// fireEvent.change(screen.getByTestId('switch1').querySelector('input'),{
//   target: { value: true }
// })

// fireEvent.click(screen.getByTestId('switch1'),)
// /* ======================= */

// // await delay(1000) /* wait till setStore apply affects(note:this is necessary so component gets updated) */
// await waitFor(() => { }, { timeout: 1000 })
// // let targetUserId
// // await waitFor(() => {
// // const targetUserId = helpers.idShrinker(Object.keys(helpers.getStore('confirmedSales'))[0])
// expect(screen.queryByText(targetUserId)).not.toBeInTheDocument()
// // }, { timeout: 1000 })
// /* ======== */
// /* dont rm these for now [aksjd3u2] */
// // await gif.setStore_({ states_sales_onlySpecificItems: false },1000) //ok with redux-promise
// // await helpers.setStore({ states_sales_onlySpecificItems: false }) //fails
// // await gif.setStore__({ states_sales_onlySpecificItems: false },1000) //ok with redux-promise
// /* ======== */
// /* ok */
// helpers.setStore({ states_sales_onlySpecificItems: true }) //ok normal
// // await delay(1000) /* wait till setStore apply affects(note:this is necessary so component gets updated) */
// // helpers.resetStore() //ok normal

// // await delay(1000) /* wait till setStore apply affects(note:this is necessary so component gets updated) */
// /* ======== */
// // console.log(222, helpers.getStore('states_sales_onlySpecificItems'))
// // await delay(100) /* ksjdhf2ui3 */
// console.log(targetUserId)
// // screen.debug()
// // await waitForDomChange()
// await waitFor(() => {
//   screen.debug(screen.queryByText(targetUserId))
//   expect(screen.queryByText(targetUserId)).toBeInTheDocument()
// }, { timeout: 1000 })
// // screen.debug(screen.getByTestId('switch1'))
// // screen.debug(screen.queryByText(targetUserId))

// const asd = screen.queryByText('mzq6')
// screen.debug()
// expect(screen.queryByText('mzq6')).not.toBeInTheDocument()
// screen.debug(screen.queryByText(targetUserId))
/* this also works
expect(() => getByText('your text')).toThrow('Unable to find an element');
 */

// test('test redux configurations and middlewares', async () => {

//   // const asd=
//   // await helpers.setStore({ states_sales_onlySpecificItems: true })

//   // const firstAction: any = () => {
//   //   return async (dispatch) => {
//   //     const response: Promise<unknown> = dispatch({
//   //       type: 'states_sales_onlySpecificItems',
//   //       payload: fakeFetch(),
//   //     })
//   //     response.then((data) => {
//   //       return data
//   //     })
//   //   }
//   // }

//   // const promiseLogin = () => ({
//   //   type: 'states_sales_onlySpecificItems',
//   //   payload: {
//   //     promise: fakeFetch().then(i=>i),
//   //   }
//   // })

//   // const asdasdf:any='states_sales_onlySpecificItems_FULFILLED'
//   // await store.dispatch(firstAction)
//   // console.log(2, helpers.getStore('states_sales_onlySpecificItems' as any))
//   // // console.log(2, store.getState())
//   // // store.dispatch({ type: 'states_sales_onlySpecificItems', payload: false })
//   // // console.warn(2,helpers.getStore('states_sales_onlySpecificItems'))

//   // const states_sales_limitedItems = helpers.getStore('states_sales_limitedItems')
//   // console.log(111, helpers.getStore('states_sales_limitedItems').length)

//   // const asd = () => {
//   //   let res = []
//   //   for (let index = 0; index < 100000; index++) {
//   //     res = [...res, ...states_sales_limitedItems]
//   //   }
//   //   return res
//   // }

//   // gif.setStore_({ states_sales_limitedItems: asd() })
//   // console.log(222, helpers.getStore('states_sales_limitedItems').length)
//   // gif.setStore_({ states_sales_onlySpecificItems: true })

// });

/* =========================== */
// import 'react-native'
// import React from 'react'
// import {fireEvent, render} from '@testing-library/react-native'
// import Login from '../src/components/Login'
// import {expect, it, jest} from '@jest/globals'

// it('renders correctly', async () => {
//   const username = 'hi'
//   const password = 'qwerty1234'
//   let submittedData = {}
//   // @ts-ignore
//   const handleSubmit = jest.fn(data => (submittedData = data))
//   const {getByText, getByPlaceholderText} = render(
//     <Login onSubmit={handleSubmit} />,
//   )
//   const button = getByText(/submit/i)

//   await fireEvent.changeText(getByPlaceholderText(/username/i), username)
//   await fireEvent.changeText(getByPlaceholderText(/password/i), password)
//   fireEvent.press(button)

//   expect(submittedData).toEqual({password, username})
//   expect(handleSubmit).toHaveBeenCalledWith({password, username})
//   expect(handleSubmit).toHaveBeenCalledTimes(1)
// })
/* =========================== */
// import 'react-native'
// import React from 'react'
// import EasyButton from '../src/components/EasyButton'
// import {render} from '../src/test/test-utils'
// import {expect, it} from '@jest/globals'
// import {ReactTestInstance} from 'react-test-renderer'

// it('renders with the light styles for the light theme', () => {
//   const {getByText} = render(<EasyButton>Click me!</EasyButton>)
//   const innerText = getByText(/click/i)
//   const button = innerText.parent as ReactTestInstance

//   expect(button.props.style).toMatchObject({backgroundColor: 'white'})
//   expect(innerText.props.style).toMatchObject({color: 'black'})
// })

// it('renders with the dark styles for the dark theme', () => {
//   const {getByText} = render(<EasyButton>Click me!</EasyButton>, {
//     theme: 'dark',
//   })
//   const innerText = getByText(/click/i)
//   const button = innerText.parent as ReactTestInstance

//   expect(button.props.style).toMatchObject({backgroundColor: 'black'})
//   expect(innerText.props.style).toMatchObject({color: 'white'})
// })
/* =========================== */

// const onPressMock = jest.fn();
// const eventData = {
//   nativeEvent: {
//     pageX: 20,
//     pageY: 30,
//   },
// };

// const { getByText } = render(
//   <View>
//     <TouchableOpacity onPress={onPressMock}>
//       <Text>Press me</Text>
//     </TouchableOpacity>
//   </View>
// );

// fireEvent.press(getByText('Press me'), eventData);
// expect(onPressMock).toHaveBeenCalledWith(eventData);
/* =========================== */
/* =========================== */
/* =========================== */
// const onEventMock = jest.fn();
//   const { getByPlaceholderText } = render(
//     // MyComponent renders TextInput which has a placeholder 'Enter details'
//     // and with `onChangeText` bound to handleChangeText
//     <MyComponent handleChangeText={onEventMock} />
//   );

//   fireEvent(getByPlaceholderText('change'), 'onChangeText', 'ab');
//   expect(onEventMock).toHaveBeenCalledWith('ab');
/* =========================== */

// const onBlurMock = jest.fn();

// const { getByPlaceholderText } = render(
//   <View>
//     <TextInput placeholder="my placeholder" onBlur={onBlurMock} />
//   </View>
// );
/* =========================== */
/* kajshdi2u3 */
// type Props = {|
//   +dummyID?: string,
// |};

// const TextPress = ({ dummyID }: Props) => (
//   <View testID={dummyID}>
//     <Text testID="text-button">Press me</Text>
//   </View>
// );

// test('shallow rendering React elements', () => {
//   const { output } = shallow(<TextPress dummyID="2" />);

//   expect(output).toMatchSnapshot();
// });

// test('shallow rendering React Test Instance', () => {
//   const { getByTestId } = render(<TextPress />);
//   const { output } = shallow(getByTestId('text-button'));

//   expect(output).toMatchSnapshot();
// });
/* =========================== */
/* mock a whole library and use one of its functions */
/*   import { myFunction } from "./library";
  jest.mock("./library");

  const mockMyFunction = myFunction as jest.MockedFunction<typeof myFunction>;
  expect(mockMyFunction.mock.calls[0][0]).toBe(42) */
/* =========================== */
