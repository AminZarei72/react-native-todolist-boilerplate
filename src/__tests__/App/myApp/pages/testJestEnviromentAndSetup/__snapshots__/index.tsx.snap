// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fake a login form 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 24,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#9e9ef8",
          "borderRadius": 12,
          "justifyContent": "center",
          "margin": 6,
          "padding": 6,
        }
      }
    >
      <Text>
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`test a flatlist and a scrolling affect  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 16,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        "Pizza",
        "Burger",
        "Risotto",
        "French Fries",
        "Onion Rings",
        "Fried Shrimps",
        "Water",
        "Coke",
        "Beer",
        "Cheese Cake",
        "Pancakes",
        "The Impossible Burger",
        "Fanta",
        "French Omelette",
        "Onion Fries",
        "Nep Shrimps",
        "Soda",
        "Cheesy Mushroom",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    progressViewOffset={100}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={100}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="flat-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Pizza
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Burger
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Risotto
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            French Fries
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Onion Rings
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Fried Shrimps
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Water
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Coke
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Beer
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Cheese Cake
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Pancakes
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            The Impossible Burger
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Fanta
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            French Omelette
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Onion Fries
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Nep Shrimps
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Soda
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f9c2ff",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Cheesy Mushroom
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "paddingVertical": 12,
      }
    }
  >
    <Text />
  </View>
</RCTSafeAreaView>
`;
